docs/03-pages/01-building-your-application/01-routing/05-custom-app.mdx

```diff
@@ -36,6 +36,52 @@ The `Component` prop is the active `page`, so whenever you navigate between rout
 > - If your app is running and you added a custom `App`, you'll need to restart the development server. Only required if `pages/_app.js` didn't exist before.
 > - `App` does not support Next.js [Data Fetching methods](/docs/pages/building-your-application/data-fetching) like [`getStaticProps`](/docs/pages/building-your-application/data-fetching/get-static-props) or [`getServerSideProps`](/docs/pages/building-your-application/data-fetching/get-server-side-props).
 
+### Accidentally opting out from Server-Side Rendering (SSR)
+
+When you use a custom `App` component, you must be careful with conditional logic and the rendering of `Component`. For example:
+
+```tsx filename="pages/_app.tsx"
+import type { AppProps } from 'next/app'
+import { useEffect, useState } from 'react'
+
+export default function MyApp({ Component, pageProps }: AppProps) {
+  const [isServer, setIsServer] = useState(true)
+
+  useEffect(() => {
+    setIsServer(false)
+  }, [])
+
+  // This may not be what you want!
+  // It depends on your use case!
+  if (isServer) return <div>Loading...</div>
+  // The page is only rendered in the client
+  return <Component {...pageProps} />
+}
+```
+
+In this case, `isServer` is only set to false in the client. That means that on the server, only a div with the text, `Loading...` is rendered.
+
+On the client, after hydration takes place, the page `Component` will be rendered.
+
+In other words, the page component associated with the current route, is not rendered in the server.
+
+When inspecting the initial HTML of the page, you'll see the `Loading...` div element.
```

Comment:
```suggestion
When inspecting the initial HTML of the page, you'll see the `Loading...` `<div>` element.
```

Reply:
```suggestion
When inspecting the initial HTML of the page, you'll see the `Loading...` div element.
```

Reply:
I had missed these suggestions, will try to catch up asap